enigma-js/test/Enigma.spec.js

```diff
@@ -1372,53 +1372,84 @@ describe('Enigma tests', () => {
-      expect(task.receipt).toBeTruthy();
-      expect(task.transactionHash).toBeTruthy();
-      expect(task.taskId).toBeTruthy();
-      expect(task.ethStatus).toEqual(1);
-      expect(task.proof).toBeFalsy();
-      expect(endingContractBalance - startingContractBalance).toEqual(task.gasLimit * task.gasPx);
+      const gasUsed = 25;
+      const optionalEthereumData = '0x';
+      const optionalEthereumContractAddress = '0x0000000000000000000000000000000000000000';
+      stateDeltaHash = '0x0000000000000000000000000000000000000000000000000000000000000000';
+      const proof = utils.hash([
+        codeHash, task.inputsHash, initStateDeltaHash, stateDeltaHash, outputHash,
+        JSBI.BigInt(gasUsed).toString(16).padStart(16, '0'), optionalEthereumData, optionalEthereumContractAddress,
+        '0x01']);
+      const workerParams = await enigma.getWorkerParams(task.creationBlockNumber);
+      const selectedWorkerAddr = (await enigma.selectWorkerGroup(task.scAddr, workerParams, 1))[0];
+      let worker = await enigma.admin.findBySigningAddress(selectedWorkerAddr);
+      const priv = data.workers.find((w) => w[0] === selectedWorkerAddr.toLowerCase())[4];
+      const sig = EthCrypto.sign(priv, proof);
+      const startingWorkerBalance = worker.balance;
+      const startingSenderBalance = parseInt(await enigma.tokenContract.methods.balanceOf(task.sender).call());
+      let sampleContractInt = parseInt(await sampleContract.methods.stateInt().call());
+      let sampleContractBool = await sampleContract.methods.stateBool().call();
+      expect(sampleContractInt).toEqual(5);
+      expect(sampleContractBool).toEqual(true);
+      const result = await new Promise((resolve, reject) => {
+        enigma.enigmaContract.methods.commitReceipt(scAddr, task.taskId, stateDeltaHash, outputHash,
+          optionalEthereumData,
+          optionalEthereumContractAddress, gasUsed, sig).send({
+          from: worker.account,
+        }).on('receipt', (receipt) => resolve(receipt)).on('error', (error) => reject(error.message));
+      });
+      sampleContractInt = parseInt(await sampleContract.methods.stateInt().call());
+      sampleContractBool = await sampleContract.methods.stateBool().call();
+      expect(sampleContractInt).toEqual(5);
+      expect(sampleContractBool).toEqual(true);
+      worker = await enigma.admin.findBySigningAddress(selectedWorkerAddr);
+      const endingWorkerBalance = worker.balance;
+      const endingSenderBalance = parseInt(await enigma.tokenContract.methods.balanceOf(task.sender).call());
+      expect(endingWorkerBalance - startingWorkerBalance).toEqual(gasUsed * task.gasPx);
+      expect(endingSenderBalance - startingSenderBalance).toEqual((task.gasLimit - gasUsed) * task.gasPx);
+      expect(result.events.ReceiptVerified).toBeTruthy();
+    });
+
+    it('should count state deltas', async () => {
+      const count = await enigma.admin.countStateDeltas(scAddr);
+      expect(count).toEqual(1);
```

Comment:
I would add a check to verify that the number of the contract's deltas hasn't changed after the call to `compute`

Reply:
resolved: c0ddcde83e6a04514cd1e05611303c2d1dc1dbcf

Reply:
will resolve the additional task verification comment in a following PR as per our discussion offline!